### DISCUSSION TITLE
Crash when mouse move on Osu!Lazer window

### DISCUSSION THREAD

**Discussion by LIII-Works** [ppy/osu#27289] [Q&A] [answered]
### Type

Crash to desktop

### Bug description

The game crashes shortly after launching when the mouse is on osu!Lazer window. If there is another window on top of osu!Lazer and the mouse are on that window, osu wont crash

### Screenshots or videos

https://streamable.com/dok9om

### Version

2024.219.0

### Logs

[1708460748.updater.log](https://github.com/ppy/osu/files/14350841/1708460748.updater.log)
[1708460748.input.log](https://github.com/ppy/osu/files/14350842/1708460748.input.log)
[1708460748.network.log](https://github.com/ppy/osu/files/14350843/1708460748.network.log)
[1708460748.performance.log](https://github.com/ppy/osu/files/14350844/1708460748.performance.log)
[1708460748.runtime.log](https://github.com/ppy/osu/files/14350845/1708460748.runtime.log)

**Comment by bdach** [comment]
Please try disabling shadowplay if you have it enabled.

**Reply by LIII-Works** [reply, marked as answer]
Thank you for the suggestion. 

I have tested my overlay applications and found that the crash was caused by RTSS `Guru3D RTSS Rivatuner Statistics Server`. Upon exit RTSS, osu!lazer work as intended.

Will this be dealt with as a bug? or this behaviour is intended?

**Reply by bdach** [reply]
I'm not sure. I'm not familiar with this particular piece of what I presume to be OEM software.

If you can attach a windows event viewer trace of what the crash was (because the logs don't contain any, meaning that it was something lower down), then maybe we can debug further - but I fully expect to see the ususal "access violation exception" spiel meaning we'd have to have your exact software/hardware setup to debug this.

**Reply by LIII-Works** [reply]
Im fine with providing logs and details, can you kindly provide me a list of what you need?

**Reply by bdach** [reply]
Please attach Windows Event Viewer info by performing the following steps:
1. Type `Event viewer` in the Windows search bar (shortcut: <kbd>Windows</kbd>+<kbd>S</kbd>) and press <kbd>Enter</kbd>, or open the *Run...* button (<kbd>Windows</kbd>+<kbd>R</kbd>), type in `eventvwr` and press <kbd>Enter</kbd>.
2. The *Event viewer* window will appear. Expand *Windows Logs* on the left and choose *Application*. It might take a while to load.
3. In the *Actions* pane on the right click *Save All Events As...*, then choose a location to save to.
4. On the *Display Information* popup, select the second option, then the checkbox to show all available languages, then *English (UK)* (US would also do fine).
5. Append the resulting `.evtx` file to this issue.

**Reply by LIII-Works** [reply]
[Events.zip](https://github.com/ppy/osu/files/14358225/Events.zip)

It seems there is a lot of entries in the .evtx. Let me know if you want me to trigger the crash few more times to make it easier to find in .evtx.

**Reply by bdach** [reply]
Yeah that's a dead end unfortunately.

```
Application: osu!.exe
CoreCLR Version: 8.0.224.6711
.NET Version: 8.0.2
Description: The process was terminated due to an unhandled exception.
Exception Info: System.AccessViolationException: Attempted to read or write protected memory. This is often an indication that other memory is corrupt.
Stack:
   at Veldrid.D3D11.D3D11GraphicsDevice.SwapBuffersCore(Veldrid.Swapchain)
   at osu.Framework.Platform.GameHost.DrawFrame()
   at osu.Framework.Threading.GameThread.processFrame()
   at osu.Framework.Threading.GameThread.<createThread>g__runWork|66_0()
   at System.Threading.ExecutionContext.RunInternal(System.Threading.ExecutionContext, System.Threading.ContextCallback, System.Object)
```

I don't think we're going to be able to resolve this on our end easily. What you _could_ try is changing renderer to opengl (legacy) and seeing if it helps stop that rivatuner thing from messing with the game, but I dunno.

**Reply by LIII-Works** [reply]
Changing to OpenGL // OpenGL(Legacy) works just fine with RTSS. Thank you.

If this could not be easily resolved, I would like to suggest adding this information`RTSS may cause the osu to crash when using Direct3D 11` to crash related information so other players with the same problem would know.

**Comment by smoogipoo** [comment]
It's going to lead to worse performance, so I don't suggest keeping it this way, but can you try testing with the environment variable `OSU_GRAPHICS_STAGING_BUFFER_TYPE=0`, and seeing if it fixes with that overlay active?

**Reply by LIII-Works** [reply]
![image](https://github.com/ppy/osu/assets/56265123/31bc1eae-49ee-4c4e-9599-371abb49d00b)

If im doing it right, doing `OSU_GRAPHICS_STAGING_BUFFER_TYPE=0` did not resolve the problem. I will stick with DX11 with RTSS disabled.

**Comment by AlexUnwinder** [comment]
Just installed Lazer 2024.221.0 and tried to reproduce it on the most recent RTSS 7.3.5 with no luck. Did you use the same RTSS version?
